registry.go

```
@@ -0,0 +1,131 @@
+// Returns the project as it is stored inside Harbor or an error,
+// if the project could not be created.
+func (c *RegistryRESTClient) NewRegistry(ctx context.Context, name, registryType, url string,
+	credential *model.RegistryCredential, insecure bool) (*model.Registry, error) {
+	rReq := &model.Registry{
+		Credential: credential,
+		Insecure:   insecure,
+		Name:       name,
+		Type:       registryType,
+		URL:        url,
+	}
+
+	_, err := c.parent.Client.Products.PostRegistries(
+		&products.PostRegistriesParams{
+			Registry: rReq,
+			Context:  ctx,
+		}, c.parent.AuthInfo)
+
+	err = handleSwaggerRegistryErrors(err)
+	if err != nil {
+		return nil, err
+	}
+
+	registry, err := c.Get(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	return registry, nil
+}
+
+// Get returns a project identified by name.
+// Returns an error if it cannot find a matching project or when
+// having difficulties talking to the API.
+func (c *RegistryRESTClient) Get(ctx context.Context, name string) (*model.Registry, error) {
+	if name == "" {
+		return nil, errors.New("no name provided")
+	}
+	resp, err := c.parent.Client.Products.GetRegistries(
+		&products.GetRegistriesParams{
+			Name:    &name,
+			Context: ctx,
+		}, c.parent.AuthInfo)
+
+	err = handleSwaggerRegistryErrors(err)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range resp.Payload {
+		if r.Name == name {
+			return r, nil
+		}
+	}
+
+	return nil, NewRegistryError(ErrRegistryNotFoundMsg, -1, name)
+}
+
+// Delete deletes a registry.
+// Returns an error when no matching registry is found or when
+// having difficulties talking to the API.
+func (c *RegistryRESTClient) Delete(ctx context.Context,
+	r *model.Registry) error {
+	if r == nil {
+		return errors.New("no registry provided")
+	}
+
+	registry, err := c.Get(ctx, r.Name)
+	if err != nil {
+		return err
+	}
+
+	if r.ID != registry.ID {
+		return NewRegistryError(ErrRegistryMismatchMsg, r.ID, r.Name)
```

It's better to return the custom error `&ErrRegistryMismatch{}`